Node library "Error: Cannot find module"

### Current Behavior

I create a workspace with an express app and add a node library, import my library from my express app, and node is not able to find the module.

### Expected Behavior

Should be able to find the module of my library

### GitHub Repo

_No response_

### Steps to Reproduce

1. `yarn create nx-workspace`
```
Where would you like to create your workspace?: org
Which stack do you want to use?: Node
What framework should be used?: Express
Integrated monorepo, or standalone project?: Integrated monorepo
Application name: org-api
Would you like to generate a Dockerfile?: Yes
Enable distributed caching to make your CI faster: No
```

2. `yarn nx g @nx/node:library --name=utils --buildable=true --directory=libs/utils --importPath=@myorg/utils`

3. _Edited apps/org-api/src/main.ts_:
```typescript
// ...
import { utilsUtils } from '@myorg/utils';
// ...
app.get('/', (req, res) => {
  res.send({ message: utilsUtils() });
});
// ...
```

4. `yarn nx run org-api:serve:development`
```
Error: Cannot find module '/Users/victor/Development/org/dist/libs/utils/utils'
Require stack:
- /Users/victor/Development/org/dist/apps/org-api/apps/org-api/src/main.js
- /Users/victor/Development/org/dist/apps/org-api/main.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Function.Module._resolveFilename (/Users/victor/Development/org/tmp/org-api/main-with-require-overrides.js:41:36)
    at Function.Module._load (node:internal/modules/cjs/loader:922:27)
    at Function.Module._load (/Users/victor/Development/org/node_modules/@nx/js/src/executors/node/node-with-require-overrides.js:15:31)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.<anonymous> (/Users/victor/Development/org/apps/org-api/src/main.ts:1:28)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
```

### Nx Report

```shell
Node   : 18.17.0
   OS     : darwin-x64
   yarn   : 1.22.19
   
   nx                 : 16.7.4
   @nx/js             : 16.7.4
   @nx/jest           : 16.7.4
   @nx/linter         : 16.7.4
   @nx/workspace      : 16.7.4
   @nx/devkit         : 16.7.4
   @nx/esbuild        : 16.7.4
   @nx/eslint-plugin  : 16.7.4
   @nx/node           : 16.7.4
   @nrwl/tao          : 16.7.4
   typescript         : 5.1.6
```


### Failure Logs

_No response_

### Package Manager Version

_No response_

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

--- a/packages/node/src/generators/application/application.spec.ts
+++ b/packages/node/src/generators/application/application.spec.ts
@@ -51,9 +51,13 @@ describe('app', () => {
                 },
               },
               "defaultConfiguration": "development",
+              "dependsOn": [
+                "build",
+              ],
               "executor": "@nx/js:node",
               "options": {
                 "buildTarget": "my-node-app:build",
+                "runBuildTargetDependencies": false,
               },
             },
           },
@@ -235,9 +239,13 @@ describe('app', () => {
                 },
               },
               "defaultConfiguration": "development",
+              "dependsOn": [
+                "build",
+              ],
               "executor": "@nx/js:node",
               "options": {
                 "buildTarget": "my-node-app:build",
+                "runBuildTargetDependencies": false,
               },
             },
           },

--- a/packages/node/src/generators/application/application.ts
+++ b/packages/node/src/generators/application/application.ts
@@ -134,8 +134,14 @@ function getServeConfig(options: NormalizedSchema): TargetConfiguration {
   return {
     executor: '@nx/js:node',
     defaultConfiguration: 'development',
+    // Run build, which includes dependency on "^build" by default, so the first run
+    // won't error out due to missing build artifacts.
+    dependsOn: ['build'],
     options: {
       buildTarget: `${options.name}:build`,
+      // Even though `false` is the default, set this option so users know it
+      // exists if they want to always run dependencies during each rebuild.
+      runBuildTargetDependencies: false,
     },
     configurations: {
       development: {

